### PR TITLE
RHIROS-733 | Cloud provider check for Update events

### PR DESCRIPTION
**Pull request by saltgen**
Changes:
1. Added an assert for cloud provider value for  update events
2. Added unittest

Additional:
- NA

### CODE REVIEW

File: ros/processor/inventory_events_consumer.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def host_create_update_events(self, msg):
         if (
                 msg['platform_metadata'] is None
                 and msg['type'] == 'updated'
+                and msg['host']['system_profile']['cloud_provider']
         ) or 'is_ros' in msg['platform_metadata']:
```
Comment (patilsuraj767) [inline ros/processor/inventory_events_consumer.py:135]: Due to this "or" condition there can be one edge case in which hosts(VM, baremetal) will get processed. 
If user had run ros ansible playbook on baremetal host than puptoo will set `is_ros=True` for that host's insights report and ros-backend will process it.
Might be I am overthinking but possible.

Comment (saltgen) [reply ros/processor/inventory_events_consumer.py:135]: @patilsuraj767 In that case wouldn't that be a putpoo thing to sort out?

Comment (patilsuraj767) [reply ros/processor/inventory_events_consumer.py:135]: Yes, this can be fixed from puptoo side.
Approving the PR. Thanks @saltgen

Comment (patilsuraj767) [reply ros/processor/inventory_events_consumer.py:135]: One more thing, I remembered in the past @kgaikwad mentioned for few **update** message like update of display_name full host object is not passed in kafka topic. In that case this `msg['host']['system_profile']['cloud_provider']` will return null.

Comment (saltgen) [reply ros/processor/inventory_events_consumer.py:135]: Checking, thanks @patilsuraj767

Comment (saltgen) [reply ros/processor/inventory_events_consumer.py:135]: Couldn't look into this issue yet, cause of an existng issue with host-inventory, would be great if someone can help me with the verification and changes if any, thanks!

**Review by patilsuraj767 — APPROVED** [review]
I have verified the display_name update event from inventory it do sent the cloud_provider details. http://pastebin.test.redhat.com/1077393

I am approving the PR. @kgaikwad @saltgen feel free to merge it whenever we need to push this to stage.

**Comment by saltgen** [comment]
> I have verified the display_name update event from inventory it do sent the cloud_provider details. http://pastebin.test.redhat.com/1077393
> 
> I am approving the PR. @kgaikwad @saltgen feel free to merge it whenever we need to push this to stage.

Thanks! @patilsuraj767 :rocket:

**Pull request merged**